File: analysis/import/SCMLogParser/src/main/java/de/maibornwolff/codecharta/importer/scmlogparser/input/metrics/AbsoluteCodeChurn.java
package de.maibornwolff.codecharta.importer.scmlogparser.input.metrics;

import de.maibornwolff.codecharta.importer.scmlogparser.input.Modification;

public class AbsoluteCodeChurn implements Metric {
    private long accumulatedNumberOfLinesAdded = 0;
    private long accumulatedNumberOfLinesDeleted = 0;

    @Override
    public String description() {
        return "Absolute Code Churn: Sum of number of added and deleted lines of all commits for this file.";
    }

    @Override
    public String metricName() {
        return "abs_code_churn";
    }

    @Override
    public void registerModification(Modification modification) {
        accumulatedNumberOfLinesAdded += modification.getAdditions();
        accumulatedNumberOfLinesDeleted += modification.getDeletions();
    }

    @Override
    public Number value() {
        return accumulatedNumberOfLinesAdded + accumulatedNumberOfLinesDeleted;
    }
}

File: analysis/import/SCMLogParser/src/main/kotlin/de/maibornwolff/codecharta/importer/scmlogparser/input/metrics/AbsoluteCodeChurn.kt
package de.maibornwolff.codecharta.importer.scmlogparser.input.metrics

import de.maibornwolff.codecharta.importer.scmlogparser.input.Modification

class AbsoluteCodeChurn : Metric {
    private var accumulatedNumberOfLinesAdded: Long = 0
    private var accumulatedNumberOfLinesDeleted: Long = 0
    override fun description(): String {
        return "Absolute Code Churn: Sum of number of added and deleted lines of all commits for this file."
    }

    override fun metricName(): String {
        return "abs_code_churn"
    }

    override fun registerModification(modification: Modification) {
        accumulatedNumberOfLinesAdded += modification.additions
        accumulatedNumberOfLinesDeleted += modification.deletions
    }

    override fun value(): Number {
        return accumulatedNumberOfLinesAdded + accumulatedNumberOfLinesDeleted
    }
}
